xcodeproj/internal/bazel_integration_files/calculate_output_groups.py

```diff
@@ -96,49 +131,95 @@ def _calculate_guid_target_ids(base_objroot):
         with open(target_file, encoding = "utf-8") as f:
             target_pif = json.load(f)
 
-        guid = target_pif["guid"]
-        guid_target_ids[guid] = [
-            value
-            for configuration in target_pif["buildConfigurations"]
-            for key, value in configuration["buildSettings"].items()
-            if key.startswith("BAZEL_TARGET_ID")
-        ]
+        build_target_ids = {"key": "BAZEL_TARGET_ID"}
+        compile_target_ids = {"key": "BAZEL_COMPILE_TARGET_ID"}
+        for configuration in target_pif["buildConfigurations"]:
+            for key, value in configuration["buildSettings"].items():
+                if key.startswith("BAZEL_TARGET_ID"):
+                    build_target_ids[_platform_from_build_key(key)] = value
+                elif key.startswith("BAZEL_COMPILE_TARGET_ID"):
+                    compile_target_ids[_platform_from_compile_key(key)] = value
+
+        target_ids = {
+            "build": build_target_ids,
+        }
+        if len(compile_target_ids) > 1:
+            target_ids["buildFiles"] = compile_target_ids
+
+        guid_target_ids[target_pif["guid"]] = target_ids
 
     os.makedirs(guid_target_ids_parent, exist_ok = True)
     with open(guid_target_ids_path, "w", encoding = "utf-8") as f:
         json.dump(guid_target_ids, f)
 
     return guid_target_ids
 
+def _platform_from_build_key(key):
+    if key.startswith("BAZEL_TARGET_ID[sdk="):
+        return key[20:-2]
+    return ""
+
+def _platform_from_compile_key(key):
+    if key.startswith("BAZEL_COMPILE_TARGET_ID[sdk="):
+        return key[28:-2]
+    return ""
+
+def _select_target_id(target_ids, platform):
+    key = target_ids["key"]
+
+    platforms = {platform: None}
+
+    # We need to try other similar platforms (i.e. other simulator platforms if
+    # `platform`` is for a simulator). This is to support schemes with targets
+    # of multiple platforms in them. Because `dict` is insertion ordered,
+    # `platform` will be checked first.
+    platforms.update(_similar_platforms(platform))
```

Comment:
This is the only thing that is questionable. I'm not sure it matches Xcode's order exactly. In practice it won't break for autogenerated schemes. For custom schemes, there is a possibility that mixed device/simulator targets are put together, but then the default handling below should work. I can't think of ways this breaks, but it's the part that would break if anything.